src/tools/pkcs11-tool.c

```diff
@@ -2651,7 +2651,10 @@ derive_key(CK_SLOT_ID slot, CK_SESSION_HANDLE session, CK_OBJECT_HANDLE key)
 	CK_BBOOL false = FALSE;
 	CK_OBJECT_HANDLE newkey = 0;
 	CK_RV rv;
+	CK_ECDH1_DERIVE_PARAMS ecdh_parms;
 	int fd, r;
+	unsigned char* buf = NULL;  /* buffer for handling input data */
```

Comment:
what's the reason for using an allocated buffer instead of a static buffer?

what's the reason for making the buffer global in `derive_key` instead of leaving it in the `CKM_ECDH1_DERIVE` case?